src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -223,6 +252,9 @@ - (void)_handleResubscriptionNeeded
 
     _state = MTRDeviceStateUnknown;
 
+    // reset unreported events to avoid sending noncontiguous set of events
+    _unreportedEvents = nil;
```

Comment:
This part does not make sense to me.  Why are we doing that?

Reply:
You are right! Deleting this.